File: Programming-languages-and-laboratory/Python/Exercises/43_compito_java_20230221/Paziente.py
class Paziente:
    def __init__(self, codice, nome):
        self._codice = codice
        self._nome = nome
        self._visite = []
        self._visite_totali = 0

    def add_visita(self, vis):
        self._visite.append(vis)

    def incrementa_n_visite_totali(self, vt):
        self._visite_totali += vt

    def get_codice_paziente(self):
        return self._codice

    def get_nome_paziente(self):
        return self._nome

    def get_visite_totali(self):
        return self._visite_totali

    def __str__(self):
        return str(self._codice) + "\t" + self._nome + "\t" + str(len(self._visite)) + "\t" + str([str(v) for v in self._visite])

File: II - Linguaggi di programmazione e laboratorio/Java/Exercises/43_compito_java_20230221/Paziente.java
import java.util.*;

public class Paziente {
    protected int codice_paziente;
    protected String nome_paziente;
    protected List<Visita> visite;
    protected int n_visite_totali;

    public Paziente(int codice_paziente, String nome_paziente) {
        this.codice_paziente = codice_paziente;
        this.nome_paziente = nome_paziente;
        visite = new LinkedList<Visita>();
        n_visite_totali = 0;
    }

    public void addVisita(Visita vis) {
        visite.add(vis);
    }

    public void incrementaNumeroVisiteTotali(int n_visite) {
        n_visite_totali += n_visite;
    }

    public int getNumeroVisiteTotali() {
        return n_visite_totali;
    }

    public String getNomePaziente() {
        return nome_paziente;
    }

    public int getCodicePaziente() {
        return codice_paziente;
    }

    public String toString() {
        // uso medoto ".size" per ricavare dalla lista "visite" il numero di operatori
        // che hanno fatto visite
        return codice_paziente + "\t" + nome_paziente + "\t" + visite.size() + "\t" + visite;
    }
}